File: backend/src/main/java/de/nordakademie/flxplzk/transferleistungangular/domain/Product.java
package de.nordakademie.flxplzk.transferleistungangular.domain;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.persistence.Entity;

@Entity
public class Product extends RestResource {

    private String description;
    private long priceInCents;
    private boolean active;

    private Product() {
        // JPA Constructor
    }

    public Product(@JsonProperty("description") final String description,
                   @JsonProperty("priceInCents") final long priceInCents,
                   @JsonProperty("active") final boolean active) {
        this.description = description;
        this.priceInCents = priceInCents;
        this.active = active;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public long getPriceInCents() {
        return priceInCents;
    }

    public void setPriceInCents(long priceInCents) {
        this.priceInCents = priceInCents;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }
}

File: frontend/src/main/frontend/src/domain/Product.ts
export class Product {
  private _id: number;
  private _description: string;
  private _priceInCents: number;
  private _active: boolean;

  constructor(id: number, description: string, priceInCents: number, active: boolean) {
    this._id = id;
    this._description = description;
    this._priceInCents = priceInCents;
    this._active = active;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get priceInCents(): number {
    return this._priceInCents;
  }

  set priceInCents(value: number) {
    this._priceInCents = value;
  }

  get active(): boolean {
    return this._active;
  }

  set active(value: boolean) {
    this._active = value;
  }
}
